utils/message-generator/src/parser/mod.rs

```diff
@@ -9,23 +9,94 @@ use roles_logic_sv2::parsers::AnyMessage;
+///    must be present. These `CommonMessages` are `ChannelEndpointChanged`, `SetupConnection`,
+///    `SetupConnectionError`, and `SetupConnectionSuccess`. If the user wants to specify a
+///    `SetupConnection` message to be later invoked by an action, the user must specify all fields
+///    of the `SetupConnection` message and also specify a message identifier in the `"id"` field.
+///    The message identifier is a string that is a snake case representation of the message. So
+///    for `SetupConnection`, the message identifier string is `"setup_connection". Likewise, for
+///    the `SetupConnectionSuccess` message, the message identifier string is
+///    `"setup_connection_success"`.
+/// 2.
 #[derive(Debug)]
 pub enum Parser<'a> {
+    /// Stores any number or combination of `PoolMessages` (`CommonMessage`,
+    /// `JobNegotiationMessage`, `MiningMessage`, and/or `TemplateDistributionMessage`) as
+    /// specified by the `test.json` file to be later used by a specified action.
+    ///
+    /// Each value is the `PoolMessages`, and each key the message identifier so later actions can
+    /// find and use it.
     Step1(HashMap<String, AnyMessage<'a>>),
+    /// Transforms the `PoolMessages` from `Step1` in a `Sv2Frame`. The insertion of these messages
+    /// into `Sv2Frame`s is separated from `Step1` to provide the ability for the user to have
+    /// control over the frames. Specifically, being able to create a bad frame for any
+    /// `PoolMessage` and check that the test target handles the bad frame appropriately (either
+    /// expects an error or closes the connection).
+    ///
+    /// The behavior of the frame is specified in the `"frame_builders"` key value pair in the
+    /// `test.json` file. The `"frame_builders"` value is a vector of dicts, where each dict has
+    /// two key value pairs. The first key is `"type"` which can be set to `"automatic"` if the
+    /// user wants to place the `PoolMessages` into a "correct" `Sv2Frame` (the most common use),
+    /// or it can be set to `"manual"` if the user wants to construct their own `Sv2Frame`
+    /// (typically used in the case where a forced error is desired). If `"manual"` is set, the
+    /// user will need to provide the frame headers (`extension_type`, `msg_type`, `msg_length`) in
+    /// the json dict. The second key is the message identifier string, `"message_id"`, which is the
+    /// connection to the `PoolMessage` identifier, `"id"`, discussed in `Step1`.
     Step2 {
+        /// `PoolMessages` message identifier and `PoolMessages`.
         messages: HashMap<String, AnyMessage<'a>>,
+        /// `PoolMessages` message identifier and `PoolMessages` as `Sv2Frame`.
         frames: HashMap<String, Sv2Frame<AnyMessage<'a>, Slice>>,
     },
+    /// parse all the actions
+    /// we have message_ids: put all messages that you have setup to send.
+    /// i fyou want to send a message and recv + send and then send anothe rmessage, you have two
```

Comment:
*i fyou -> if you

Reply:
anothe rmessage -> another message

Reply:
Still in WIP :) would love a review after the WIP is removed though